### PR TITLE
HARVESTER: Reorganize the network configs UI

### PR DESCRIPTION
**Pull request by n313893254**
### Summary
Reorganize the network configs UI

### Occurred changes and/or fixed issues

- https://github.com/harvester/harvester/issues/2841

### Screenshot/Video

image
image

### CODE REVIEW

File: pkg/harvester/edit/network.harvesterhci.io.clusternetwork/index.vue [hunk 1]
```diff
@@ -1,19 +1,62 @@
...
 
 <template>
-  <Setting :value="value" />
+  <CruResource
+    :resource="value"
+    :mode="mode"
+    :errors="errors"
+    @finish="save"
+  >
+    <NameNsDescription
```
Comment (mantis-toboggan-md) [inline pkg/harvester/edit/network.harvesterhci.io.clusternetwork/index.vue:38]: It looks like name needs a validator to limit it to 12 characters: when I tried to create a network config I got this error 
Screen Shot 2022-09-28 at 3 02 21 PM

**Review by mantis-toboggan-md — CHANGES_REQUESTED** [review]

File: pkg/harvester/config/harvester.js [hunk 2]
```diff
@@ -500,26 +500,28 @@ export function init($plugin, store) {
...
-    NAME_COL,
-    {
-      name:     'clusterNetwork',
-      labelKey: 'harvester.network.clusterNetwork.label',
-      value:    'spec.clusterNetwork',
-      sort:     'spec.clusterNetwork',
-    },
-    AGE
-  ]);
+  configureType(HCI.VLAN_CONFIG, {
```
[the remark this answers is not in the document]
Comment (n313893254) [reply pkg/harvester/config/harvester.js:503]: The design should look like Project/namespace, users can create a cluster network. I can see the "Create Cluster Network" button.

image

[the remark this answers is not in the document]
Comment (n313893254) [reply pkg/harvester/config/harvester.js:503]: About the namespace group, sorry for I didn't check again after rebase the code. Now I hide it by adding` hiddenNamespaceGroupButton: true`. But considering namespace group whether shown in UI is default decided by `isNamespaced` which is computed by `schema.attributes.namespaced`. I think we need to find a solution to let namespace group is decided by `isNamespaced`.

File: pkg/harvester/edit/network.harvesterhci.io.vlanconfig/index.vue [hunk 3]
```diff
@@ -65,14 +67,7 @@ export default {
...
 
-      return [{
-        label: this.t('harvester.network.clusterNetwork.create'),
-        value: null,
-      }, {
-        label:    'divider',
-        disabled: true,
-        kind:     'divider'
-      }, ...out];
+      return out;
```
[the remark this answers is not in the document]
Comment (n313893254) [reply pkg/harvester/edit/network.harvesterhci.io.vlanconfig/index.vue:70]: `value.spec.clusterNetwork` of VLAN config needs query params to fill in the default value. If without these codes, cluster network will be empty.

image

File: pkg/harvester/edit/network.harvesterhci.io.clusternetwork/index.vue (hunk 1 above)
Comment (n313893254) [reply pkg/harvester/edit/network.harvesterhci.io.clusternetwork/index.vue:38]: Done

File: pkg/harvester/l10n/en-us.yaml [hunk 4]
```diff
@@ -849,7 +849,10 @@ harvester:
         label: Migratable
 
   vlanConfig:
-    title: Network Configs
+    title: Networks Configs
```
[the remark this answers is not in the document]
Comment (n313893254) [reply pkg/harvester/l10n/en-us.yaml]: Done

File: pkg/harvester/list/network.harvesterhci.io.clusternetwork.vue [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+              >
+                <i class="icon icon-actions" />
+              </button>
+            </div>
+          </div>
+        </template>
+        <template v-for="clusterNetwork in clusterNetworkWithoutConfigs" v-slot:[slotName(clusterNetwork.id)]>
+          <tr :key="clusterNetwork.id" class="main-row">
+            <td class="empty text-center" colspan="5">
+              {{ t('projectNamespaces.noNamespaces') }}
```
[the remark this answers is not in the document]
Comment (n313893254) [reply pkg/harvester/list/network.harvesterhci.io.clusternetwork.vue]: Done

File: pkg/harvester/list/network.harvesterhci.io.clusternetwork.vue [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+      const resource = this.$store.getters[`${ inStore }/byId`](HCI.CLUSTER_NETWORK, clusterNetwork);
+
+      this.$store.commit(`action-menu/show`, {
+        resources: [resource],
+        elem:      event.target
+      });
+    },
+
+    createVlanConfigLocation(group) {
+      const clusterNetwork = group.key;
```
[the remark this answers is not in the document]
Comment (n313893254) [reply pkg/harvester/list/network.harvesterhci.io.clusternetwork.vue:151]: Done

File: pkg/harvester/list/network.harvesterhci.io.clusternetwork.vue [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+    &, &::after {
+        height: 50px;
+    }
+
+    &::after {
+        right: -20px;
+    }
+
+    SPAN {
+      color: #000 !important;
```
[the remark this answers is not in the document]
Comment (n313893254) [reply pkg/harvester/list/network.harvesterhci.io.clusternetwork.vue]: Done

File: pkg/harvester/config/harvester.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (mantis-toboggan-md) [reply pkg/harvester/config/harvester.js:499]: When the list is flat, when I create a cluster network and I am redirected back to the list afterwards, I can't see the cluster network I just created. But I see projects/namespaces works exactly the same way and no one has complained so, I don't think it should stop this PR. (and I don't have any good ideas to make it better anyway)

**Review by mantis-toboggan-md — APPROVED** [review]
LGTM

**Pull request merged**